hyperlane/relayer_runner.go

```diff
@@ -163,90 +190,97 @@ func (r *RelayerRunner) checkHyperlaneTransferStatus(ctx context.Context, transf
 	return true, nil
 }
 
-// findSettlementsToRelay checks the order settlement table for any new
-// settlements that have been initiated and creates pending hyperlane transfers
-// from them
-func (r *RelayerRunner) findSettlementsToRelay(ctx context.Context) error {
-	// poll the order settler db table and check if there are any new order
-	// settlements that should be relayed
-	pendingSettlements, err := r.db.GetAllOrderSettlementsWithSettlementStatus(ctx, dbtypes.SettlementStatusSettlementInitiated)
-	if err != nil {
-		return fmt.Errorf("getting pending order settlements: %w", err)
-	}
+// RelayOpts provides users options for how the relayer should behave when
+// relaying a tx.
+type RelayOpts struct {
```

Comment:
Imo the interface exposed in this RelayOpts struct is a little overcomplicated and introduces some unnecessarily coupling between modules. I might be missing something though its a large pr, let me know if you disagree.

Profitability - In the current definition, the relayer has its own definition of profitability based on the properties of the hyperlane message and the resulting side effects, namely transfer size. This is too specific and opinionated in my opinion and the fact this doesnt serve timeouts in an intuitive way is evidence of that imo. Im not sure that the relayer calculating profit based on a percentage of transfer size even generalizes well to all settlement configurations. For example in the early stages where transfer sizes and the amount of capital solvers have is limited, settlement costs are likely to be a much larger percentage of batch sizes on ethereum than other chains such that a flat percentage might not capture the definition of profitability for settlement across all routes. Would prefer it if it simply took a flat usdc value. The settlement handler as a client of the relayer can calculate the profit from filling based on amount ins and outs for the orders in the batch and set a usdc relay cost cap accordingly that maintains some minimum margin.

Submitter - There is no need to expose the data layer in the interface like this. Clients should know if they need to call the SubmitTxToRelay function again based on the error it returns and clients should be stateful enough that they can retry without needing to create new hyperlane messages. SubmitTxToRelay should be idempotent to make this easy and return no error if its a repeat call.

Delay - This is also functionality that seems unnecessary, the client should already be handling retries in the event they delay they specify is not enough.

Reply:
making the SubmitToRelay ignore duplicate submissions https://github.com/skip-mev/skip-go-fast-solver/pull/3/commits/1bdc6834d906d6ea767c3a64709db79e28c0fba0. making the order settler submit all initiated settlement tx's to be relayed in its main loop (not on startup as discussed, this seemed better) https://github.com/skip-mev/skip-go-fast-solver/pull/3/commits/88e085e2efb212d3cfab09fa9c3300367d0ae7ff. and making the order timeout worker submit all pending refunds that have their initiate timeout txs submitted for refund https://github.com/skip-mev/skip-go-fast-solver/pull/3/commits/f0305ff2f0e2b43b08ec121789eac234444c49d3

Reply:
max batch tx fee uusdc calculation https://github.com/skip-mev/skip-go-fast-solver/pull/3/commits/416d4d69b46f87efb0fc02bc131e5ab4a2fb240f